### ISSUE TITLE
Crash due to Amazon Cloudfront returning badly-formatted TLS alerts when SNI is not sent

### ISSUE THREAD

**Issue by agu3rra** [nabla-c0d3/sslyze#437]
**Describe the bug**
Stack trace error during a regular scan at uol.com:443

**To Reproduce**
1. Install SSLyze using pipenv. Installed version 3.0.4.
2. Run the following command on the virtual env: python -m sslyze --regular uol.com:443
4. See error: log.txt attached

**Expected behavior**
Full report with no stack trace errors.

**Python environment (please complete the following information):**
 - OS: Ubuntu 20.04
 - Python version: 3.8.2

**Additional context**
Submitting the bug attaching error log as per stack trace instructions. Hoping it can be of use.

[log.txt](https://github.com/nabla-c0d3/sslyze/files/4672447/log.txt)

**Comment by nabla-c0d3** [comment]
Thanks for the log file - very helpful!

**Comment by nabla-c0d3** [comment]
The same error can be reproduced with the OpenSSL CLI:
```
$ openssl version
OpenSSL 1.1.1  11 Sep 2018

$ openssl s_client -connect uol.com:443  -noservername
CONNECTED(00000005)
140501950206400:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../ssl/record/ssl3_record.c:332:
```

The server is returning an invalid TLS record when no server name indication is passed. More specifically, it is returning a TLS alert with invalid bytes for the TLS version (\x00\x00). It looks like this is a bug in how Amazon Cloudfront's TLS implementation rejects TLS handshakes that don't use the SNI extension.

I will have to come up with some kind of workaround..

**Comment by nabla-c0d3** [comment]
Fixed in v3.0.6.
